packages/controllers/src/plugins/PluginController.ts

```diff
@@ -431,20 +434,19 @@ export class PluginController extends EventEmitter {
       throw new Error('Expected array of plugin names.');
     }
 
-    const state = this.store.getState();
-    const newPlugins = { ...state.plugins };
-    const newPluginStates = { ...state.pluginStates };
+    const newPlugins = { ...this.state.plugins };
+    const newPluginStates = { ...this.state.pluginStates };
 
-    pluginNames.forEach((name) => {
-      this._stopPlugin(name, false);
-      delete newPlugins[name];
-      delete newPluginStates[name];
+    pluginNames.forEach((pluginName) => {
+      this._stopPlugin(pluginName, false);
+      delete newPlugins[pluginName];
+      delete newPluginStates[pluginName];
     });
     this._removeAllPermissionsFor(pluginNames);
 
-    this.updateState({
-      plugins: newPlugins,
-      pluginStates: newPluginStates,
+    this.update((state: any) => {
+      state.plugins = newPlugins;
+      state.pluginStates = newPluginStates;
```

Comment:
Should all of this be moved into the `update` callback to adhere to `immer` best practices? One of the benefits of `immer` is that we never have to spread things to preserve immutability. Not sure whether side effects should go in the callback, maybe @Gudahtt knows.

 Ref: https://immerjs.github.io/immer/update-patterns